src/index.js

```diff
@@ -34,20 +36,30 @@
 
 /**
  * @public
+ * @template T
  * @typedef Response
+ * @property {number} status
+ * @property {string} statusText
  * @property {Options} config the request configuration
- * @property {any} data the decoded response body
+ * @property {T} data the decoded response body
+ * @property {Headers} headers
```

Comment:
I've added all fields that are assigned to the response to properly match implementation (dunno if it's on purpose, or we wanna strictly "match" axios response - in that case those need to be removed + omit those within implementation)